shared-bindings/usb_hid/Device.c

```diff
@@ -98,10 +98,30 @@ const mp_obj_property_t usb_hid_device_usage_obj = {
               (mp_obj_t)&mp_const_none_obj},
 };
 
+//|   .. method:: get_report(self)
+//|
+//|     Get the next out report from the queue, or ``None`` when no reports.
+//|
+STATIC mp_obj_t usb_hid_device_obj_get_report(mp_obj_t self_in) {
+    uint8_t *data;
+
+    usb_hid_device_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    if (self->out_report_count) {
+        self->out_report_count = 0;
+        data = (uint8_t *)m_malloc(1, false);
```

Comment:
How about changing this to `read_report_into` so that it doesn't do an allocation?

Reply:
That is an excellent idea. However, I wonder how it will work when we store more than one report in the buffer? I was thinking we could then just hand the allocated string to use user and remove it from the buffer without copying then, but with this we will need to copy and free it.

Reply:
Nevermind, I just realized that the string is added to the buffer in an interrupt, so the whole buffer would need to be pre-allocated anyways.